unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/UnifiedMapNoIteratorTest.java

```diff
@@ -76,6 +94,26 @@ public void RichIterable_getLast_empty_null()
         // Not applicable
     }
 
+    @Override
+    public void MapIterable_detectOptional()
+    {
+        /**
+         * TODO: {@link UnifiedMap#detectOptional(org.eclipse.collections.api.block.predicate.Predicate2)} should be optimized to not use an iterator
+         */
```

Comment:
This test fails for UnifiedMap.detectOptional(Predicate2), which unnecessarily delegates to an iterator. Note that detect(Predicate), detect(Predicate2), and detectOptional(Predicate) are all already tested and optimized.

Reply:
Issue raised: #442